### PR TITLE
Allowing Block Device Mapping in Amazonia AUTO-729

### PR DESCRIPTION
**Pull request by fuddstar**
Removed hdd_size from ASG and replaced with block device mappings
ec2/asg defaults captured in default.yaml

Updated systems and unit tests for just about everything

Currently all properties need to be passed in to block device mappings

### CODE REVIEW

File: test/unit_tests/test_yaml_complete_valid.yaml [hunk 1]
```diff
@@ -166,13 +163,20 @@ autoscaling_units:
...
         - 'autoscaling:EC2_INSTANCE_TERMINATE'
         - 'autoscaling:EC2_INSTANCE_TERMINATE_ERROR'
+      block_devices_config:
+        -
+          device_name: '/dev/xvda'
+          ebs_volume_size: '15'
+          ebs_volume_type: 'standard'
+          ebs_encrypted: False
+          ebs_snapshot_id: ''
+          virtual_name: False
```
Comment (aynsof) [inline test/unit_tests/test_yaml_complete_valid.yaml]: Ephemeral device test

File: test/unit_tests/test_stack.py [hunk 2]
```diff
@@ -59,6 +59,14 @@ def setup_resources():
...
     db_storage_type = 'gp2'
 
+    block_devices_config = [{
+        'device_name': '/dev/xvda',
+        'ebs_volume_size': '15',
+        'ebs_volume_type': 'gp2',
+        'ebs_encrypted': False,
+        'ebs_snapshot_id': '',
+        'virtual_name': False}]
+
```
Comment (aynsof) [inline test/unit_tests/test_stack.py]: Ephemeral device test

File: test/unit_tests/test_block_devices.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+         'ebs_volume_type': 'gp2',
+         'ebs_encrypted': False,
+         'ebs_snapshot_id': '',
+         'virtual_name': True},
+        {'device_name': '/dev/sda2',
+         'ebs_volume_size': '15',
+         'ebs_volume_type': 'gp2',
+         'ebs_encrypted': False,
+         'ebs_snapshot_id': '',
+         'virtual_name': False}
```
Comment (aynsof) [inline test/unit_tests/test_block_devices.py]: Ephemeral device test

File: test/sys_tests/test_sys_block_devices.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+         'ebs_encrypted': False,
+         'ebs_snapshot_id': '',
+         'virtual_name': True},
+        {'device_name': '/dev/sda2',
+         'ebs_volume_size': '15',
+         'ebs_volume_type': 'gp2',
+         'ebs_encrypted': True,
+         'ebs_snapshot_id': '',
+         'virtual_name': False}
+    ]
```
Comment (aynsof) [inline test/sys_tests/test_sys_block_devices.py]: Ephemeral device test

File: test/unit_tests/test_yaml_complete_valid.yaml (hunk 1 above)
Comment (fuddstar) [inline test/unit_tests/test_yaml_complete_valid.yaml]: Addressed

File: test/sys_tests/test_sys_block_devices.py (hunk 4 above)
Comment (fuddstar) [inline test/sys_tests/test_sys_block_devices.py]: Addressed

File: amazonia/classes/asg_config.py [hunk 5]
```diff
@@ -1,7 +1,10 @@
+#!/usr/bin/python3
+
+
 class AsgConfig(object):
     def __init__(self, sns_topic_arn, sns_notification_types, health_check_grace_period,
                  health_check_type, minsize, maxsize, image_id, instance_type, userdata,
-                 iam_instance_profile_arn, hdd_size):
+                 iam_instance_profile_arn, block_devices_config):
```
Comment (GypsyBojangles) [inline amazonia/classes/asg_config.py]: block_devices_config docstrings

Comment (fuddstar) [inline amazonia/classes/asg_config.py]: Added: List containing block device mappings

**Comment by GypsyBojangles** [comment]
Approved

**Comment by GypsyBojangles** [comment]
Approved

**Pull request merged**